client/src/Auth/Register/register.test.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {mount, shallow} from 'enzyme';
+
+import Register from './register';
+
+describe('Register', () => {
+  it('username should match inputs.username', () => {
```

Comment:
I'm confused by this. This is the Register page, I don't think there is any need to initialise the input fields to any previous values. Or is that to test that you can reset the fields? Because it's not testing that.

What this file should be doing is test that the user interaction is working as expected, e.g.:
- it('should show empty fields for the user name and password with submit button disabled and no confirm password field')
- it('should show the username error message when the username input string is shorter than 4 characters')
- it('should not show the username error message when the username input string is 4 or more characters')
etc, etc...

It's ok to only write the descriptions of the tests without the implementation in the first instance, like I did above. This page probably needs a long list of tests and I'm not too bothered if you can't write them all now. But I would really like all the expected behaviour to be described. It's documentation and it allows someone else to write the tests if they have time.

Reply:
It's also a controlled component, it's there to make sure the inputs reflect the state of the parent container. Will get on and do the rest of the tests.